AmplifyPlugins/Core/AWSPluginsCore/Model/Support/GraphQLDocumentnputValue.swift

```diff
@@ -10,7 +10,7 @@ import Foundation
 /// A container to hold either an object or a value, useful for storing document inputs and allowing manipulation at
 /// the first level of the object
 public enum GraphQLDocumentInputValue {
-    case value(Any)
+    case scalarOrString(Any)
```

Comment:
As above: `scalar` should be sufficient--all non-optional types can be safely interpolated via `String(describing:)`. However, what you really want is for the incoming value to be safely representable as a primitive type (as opposed to a nested JSON-like object), rather than relying on an internal `String(describing:)` implementation that may change without warning (hello, NSData!) so you could define some protocol, and some convenience extensions to make it trivial to pass primitives into the enum value:

```swift
public enum GraphQLDocumentInputValue {
    case value(GraphQLDocumentValueRepresentable)
    case object([String: GraphQLDocumentValueRepresentable])
}

public protocol GraphQLDocumentValueRepresentable {
    var graphQLDocumentValue: String { get }
}

extension Bool: GraphQLDocumentValueRepresentable {
    public var graphQLDocumentValue: String {
        return "\(self)"
    }
}

extension Float: GraphQLDocumentValueRepresentable {
    public var graphQLDocumentValue: String {
        return "\(self)"
    }
}

extension Int: GraphQLDocumentValueRepresentable {
    public var graphQLDocumentValue: String {
        return "\(self)"
    }
}

extension String: GraphQLDocumentValueRepresentable {
    public var graphQLDocumentValue: String {
        return self
    }
}

...etc
```

Reply:
thanks for providing the example! i added this in with a few modifications:

- Added `extension Decimal: GraphQLDocumentValueRepresentable` as we represent Double's as Decimal. from what i recall it something to do with rounding. current we don't have any scalar cases other than int and string, but left this in for now.
- `    case object([String: GraphQLDocumentValueRepresentable])` doesn't quite work since if you take `QueryPredicate` to GraphQL input for example, the value can be infinitely nested. so we're storing strings of strings of values in `[String: Any]`

Reply:
so digging into this deeper, having `.scalar(GraphQLDocumentValueRepresentable)` seems to work well in this class. Adding it to the `.object[String:GraphQLDocumentValueRepresentable]` case doesn't work as it may contain arrays as well. then it will evolve to something similar to `JSONValue`.